### PR TITLE
zephyr: Update to zephyr v2.4.0.

### PR DESCRIPTION
**Pull request by MaureenHelm**
Minor changes to the zephyr port to update to zephyr v2.4.0. 

~~This PR is in draft status until the zephyr release is finalized.~~
The PR is now ready to go.

~~@tejlmand will you please have a look at the commit `zephyr: Move system includes to the end of Z_CFLAGS.`~~
A minor zephyr build system bug was fixed and this workaround commit was removed.

### CODE REVIEW

**Comment by MaureenHelm** [comment]
Updated from zephyr v2.4.0-rc1 to -rc2 and rebased to pick up 5c503de52166f6ebd4ef59b58b130a730782823d

File: ports/zephyr/CMakeLists.txt [hunk 1]
```diff
@@ -18,7 +18,7 @@ add_custom_target(
   outputexports
   COMMAND echo CC="${CMAKE_C_COMPILER}"
   COMMAND echo AR="${CMAKE_AR}"
-  COMMAND echo Z_CFLAGS=${system_includes} ${includes} ${definitions} ${options}
+  COMMAND echo Z_CFLAGS=${includes} ${definitions} ${options} ${system_includes}
```
Comment (tejlmand) [inline ports/zephyr/CMakeLists.txt:21]: This should not be needed.
Fixed the issue here: https://github.com/zephyrproject-rtos/zephyr/pull/28584
that could result in an empty `-isystem`.

**Review by tejlmand — COMMENTED** [review]
> @tejlmand will you please have a look at the commit zephyr: Move system includes to the end of Z_CFLAGS.

Bug fix PR: zephyrproject-rtos/zephyr#28584

**Comment by MaureenHelm** [comment]
Updated from zephyr v2.4.0-rc2 to -rc3

**Review by tejlmand — APPROVED** [review]

**Comment by MaureenHelm** [comment]
Updated from zephyr v2.4.0-rc3 to the final v2.4.0 release. This PR is now ready to go

**Comment by dpgeorge** [comment]
Thank you!  Rebased and merged in ce49be43b1ac7582edcfec20c56928781a512a7d through 997ec9e8ccaf4f1d8a950ef025e098e6b033c66c

**Pull request closed without merging**